src/Bicep.Core/Analyzers/Linter/Rules/OutputsShouldNotContainSecretsRule.cs

```diff
@@ -0,0 +1,162 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Deployments.Core.Extensions;
+using Bicep.Core.Diagnostics;
+using Bicep.Core.Parsing;
+using Bicep.Core.Semantics;
+using Bicep.Core.Semantics.Namespaces;
+using Bicep.Core.Syntax;
+using Bicep.Core.TypeSystem;
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Linq;
+
+namespace Bicep.Core.Analyzers.Linter.Rules
+{
+    public sealed class OutputsShouldNotContainSecretsRule : LinterRuleBase
+    {
+        public new const string Code = "outputs-should-not-contain-secrets";
+
+        public OutputsShouldNotContainSecretsRule() : base(
+            code: Code,
+            description: CoreResources.OutputsShouldNotContainSecretsRuleDescription,
+            docUri: new Uri($"https://aka.ms/bicep/linter/{Code}")
+        )
+        {
+        }
+
+        public override string FormatMessage(params object[] values)
+        {
+            return string.Format(
+                CoreResources.OutputsShouldNotContainSecretsMessageFormat,
+                this.Description,
+                values.First());
+        }
+
+        public override IEnumerable<IDiagnostic> AnalyzeInternal(SemanticModel model)
+        {
+            var visitor = new OutputVisitor(this, model);
+            visitor.Visit(model.SourceFile.ProgramSyntax);
+            return visitor.diagnostics;
+        }
+
+        private class OutputVisitor : SyntaxVisitor
+        {
+            public List<IDiagnostic> diagnostics = new List<IDiagnostic>();
+
+            private readonly OutputsShouldNotContainSecretsRule parent;
+            private readonly SemanticModel model;
+
+            public OutputVisitor(OutputsShouldNotContainSecretsRule parent, SemanticModel model)
+            {
+                this.parent = parent;
+                this.model = model;
+            }
+
+            public override void VisitOutputDeclarationSyntax(OutputDeclarationSyntax syntax)
+            {
+                // Does the output name contain 'password' (suggesting it contains an actual password)?
+                if (syntax.Name.IdentifierName.ToLowerInvariant().Contains("password"))
```

Comment:
>.ToLowerInvariant().Contains("password") [](http://example.com/codeflow?start=46&length=40)

There's an overload that accepts a `StringComparison` that will save you from making a copy of every output symbolic name: [https://docs.microsoft.com/en-us/dotnet/api/system.string.contains?view=net-6.0#System_String_Contains_System_String_System_StringComparison](https://docs.microsoft.com/en-us/dotnet/api/system.string.contains?view=net-6.0#System_String_Contains_System_String_System_StringComparison)_

Reply:
Done, thanks.